services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -271,35 +282,79 @@ async def on_shutdown() -> None:
     app.add_event_handler("shutdown", on_shutdown)
 
 
-async def _list_repositories(app: FastAPI) -> list[str]:
-    logger.debug("listing repositories")
-    # if there are more repos, the Link will be available in the response headers until none available
-    path = f"/v2/_catalog?n={get_application_settings(app).DIRECTOR_REGISTRY_CLIENT_MAX_NUMBER_OF_RETRIEVED_OBJECTS}"
-    repos_list: list = []
-    while True:
-        result, headers = await registry_request(app, path)
-        if result["repositories"]:
-            repos_list.extend(result["repositories"])
-        if "Link" not in headers:
-            break
-        path = str(headers["Link"]).split(";")[0].strip("<>")
-    logger.debug("listed %s repositories", len(repos_list))
-    return repos_list
+def _get_prefix(service_type: ServiceType) -> str:
+    return f"{DIRECTOR_SIMCORE_SERVICES_PREFIX}/{service_type.value}/"
+
+
+_SERVICE_TYPE_FILTER_MAP: Final[dict[ServiceType, tuple[str, ...]]] = {
+    ServiceType.DYNAMIC: (_get_prefix(ServiceType.DYNAMIC),),
+    ServiceType.COMPUTATIONAL: (_get_prefix(ServiceType.COMPUTATIONAL),),
+    ServiceType.ALL: (
+        _get_prefix(ServiceType.DYNAMIC),
+        _get_prefix(ServiceType.COMPUTATIONAL),
+    ),
+}
+
+
+async def _list_repositories_gen(
+    app: FastAPI, service_type: ServiceType
+) -> AsyncGenerator[list[str], None]:
+    with log_context(_logger, logging.DEBUG, msg="listing repositories"):
+        path = f"/v2/_catalog?n={get_application_settings(app).DIRECTOR_REGISTRY_CLIENT_MAX_NUMBER_OF_RETRIEVED_OBJECTS}"
+        result, headers = await registry_request(app, path=path)  # initial call
+
+        while True:
+            if "Link" in headers:
+                next_path = str(headers["Link"]).split(";")[0].strip("<>")
+                prefetch_task = asyncio.create_task(
+                    registry_request(app, path=next_path)
+                )
+            else:
+                prefetch_task = None
+
+            yield list(
```

Comment:
such a c++ guys ;-) 
Why not using introspection here?
```python

prefix = _SERVICE_TYPE_FILTER_MAP[service_type]
yield [ r for r in result['repositories'] if  f"{r}".startswith(prefix) ]
```

Reply:
this is not a generator. but well why not using filter?